rfd/0164-scoped-rbac.md

```diff
@@ -0,0 +1,956 @@
+metadata:
+  name: requester
+spec:
+  allow:
+    request:
+      search_as_roles:
+        - access
+```
+
+The same applies to impersonation, if access list granted Alice the role `impersonator` below at scope `/env/prod/lab`, 
+Alice would be able to impersonate role `jenkins`, but only at the scope `/env/prod/lab` or a more specific one, e.g. `/env/prod/lab/cabinet-west`.
+
+```yaml
+kind: role
+version: v5
+metadata:
+  name: impersonator
+spec:
+  allow:
+    impersonate:
+      users: ['jenkins']
+      roles: ['jenkins']
+```
+
+**Note:** While it’s tempting to support scope templates, we will push this out of the scope of this RFD.
+
+#### Roles and Access Lists in resource groups
+
+A special case is when a role or an access list is assigned to a certain resource group.  
+
+Only roles that have `grantable_scope` matching the resource group can be assigned to the resource group.
+
+The same applies to access lists, the scope of the access list grants should always match the scope of the roles it grants access to and 
+should not exceed the scope of the access list itself.
+
+In both of those cases, parent resource group must be specified both for access lists and roles and should equal or be more specific than the scope it was created in:
+
+```yaml
+kind: role
+metadata:
+  name: lab-admin
+spec:
+  grantable_scope: /env/prod/lab
+  parent_resource_group: /env/prod/lab
+```
+
+```yaml
+kind: access_list
+metadata:
+  name: lab-personnel
+spec:
+  grants_scope: /env/prod/lab
+  parent_resource_group: /env/prod/lab
+```
+
+Roles created within a scope will have `grantable_scope` and `parent_resource_group` to be equal to scope, or more specific. 
+
+For example, any role created within a scope `/env/prod/lab` must have the same `grantable_scope` and `parent_resource_group`  - `/env/prod/lab` or more specific one, 
+for example `/env/prod/lab/west-wing`.
+
+The roles and access lists created in the scope must grant access to scopes equal, or more specific than the scope they were created in to prevent lateral expansion or permissions.
+
+These invariants will let us make sure that any role created within a scope, will only grant permissions in the same, or more specific scope. 
+
+We will apply the same invariants to any other resources created within a scope. 
+
+#### Scoped Join Tokens
+
+Join tokens with a `parent_resource_group` will bind any resource using this join method to the resource group specified in `parent_resource_group` or a more specific one. 
+
+```yaml
+# token.yaml
+kind: token
+version: v2
+metadata:
+  name: my-token-name
+spec:
+  parent_resource_group: '/dev'
+  roles: 
+    - Node
+    - App
+```
+
+Join tokens created by roles granted within a scope must have `parent_resource_group` equal to this scope `/dev` or a more specific scope, e.g. `/dev/lab`
+
+#### Access Requests
+
+Access requests are bound to the scope they are created within, when Alice requests access to environment `/dev/lab` with role access, 
```

Comment:
The reference here is to role based access requests and not resource access requests, is that correct?

Reply:
search-based access requests are a subset of role based, so there should be no difference, I added a section "Grants and access requests" to explain.